op-proposer/proposer/service.go

```diff
@@ -26,16 +26,20 @@ import (
 	"github.com/ethereum/go-ethereum/log"
 )
 
-var (
-	ErrAlreadyStopped = errors.New("already stopped")
-)
+var ErrAlreadyStopped = errors.New("already stopped")
 
 type ProposerConfig struct {
 	// How frequently to poll L2 for new finalized outputs
 	PollInterval   time.Duration
 	NetworkTimeout time.Duration
 
-	L2OutputOracleAddr common.Address
+	// How frequently to post L2 outputs when the DisputeGameFactory is configured
+	ProposalInterval time.Duration
+
+	L2OutputOracleAddr     common.Address
+	DisputeGameFactoryAddr *common.Address
```

Comment:
Why is one a pointer and the other a value? Aren't both optional now? I'd let both be a value or pointer, if there isn't a semantic reason for letting them be different types. Just need to compare to `(common.Address{})` for checking the zero value if using values types.